packages/repeater/src/lib/Repeater.spec.ts

```diff
@@ -126,5 +220,64 @@ describe('Repeater', () => {
         )
       ).once();
     });
+
+    it('should have RunningStatus.OFF after start() and stop()', async () => {
+      await repeater.start();
+      await repeater.stop();
+      expect(repeater.runningStatus).toBe(RunningStatus.OFF);
+    });
+
+    it('should throw an error on stop() without start()', async () => {
+      const res = repeater.stop();
+
+      await expect(res).rejects.toThrow('Cannot stop non-running repeater.');
+    });
+
+    it('should throw an error on stop() twice', async () => {
+      await repeater.start();
+      await repeater.stop();
+
+      const res = repeater.stop();
+
+      await expect(res).rejects.toThrow('Cannot stop non-running repeater.');
+    });
+
+    it('should stop() on process termination', async () => {
+      const spiedRepeater = spy(repeater);
+
+      await repeater.start();
+      process.emit('SIGTERM' as any);
```

Comment:
It is better to avoid interaction with the actual process. You can create a spy as follows:

```ts
const spiedProcess = spy(process);
```

then it is possible to mock a particular handler, however you have to store a callback to a local variable:

```ts
// arrange
let action!: () => unknown;

when(spiedProcess.on(anyString(), anything()).thenCall((_, callback) => action = callback);
// ...

// act
action();

// assert
// ...
```